File: ios/SolSnatcher/SolSnatcher/Models/Wallet.swift
import Foundation

struct Wallet: Identifiable, Codable {
    let id: String
    let pubkey: String
    let path: String
    let riskPctPerTrade: Double
    let balance: Double
    let busy: Bool
    let dailyPnL: Double
    let totalPnL: Double
}

struct Position: Identifiable, Codable {
    let id: String
    let walletId: String
    let mint: String
    let baseQty: Double
    let avgCost: Double
    let tpPct: Double
    let slPct: Double
    let trailingPct: Double
    let status: String
    let createdAt: String
    let updatedAt: String
}

struct SystemStatus: Codable {
    let wallets: Int
    let activePositions: Int
    let busyWallets: Int
    let totalBalance: Double
    let totalPnL: Double
}

File: android/app/src/main/java/com/solsnatcher/data/model/Wallet.kt
package com.solsnatcher.data.model

import com.google.gson.annotations.SerializedName

data class Wallet(
    @SerializedName("id")
    val id: String,
    
    @SerializedName("pubkey")
    val pubkey: String,
    
    @SerializedName("path")
    val path: String,
    
    @SerializedName("riskPctPerTrade")
    val riskPctPerTrade: Double,
    
    @SerializedName("balance")
    val balance: Double,
    
    @SerializedName("busy")
    val busy: Boolean,
    
    @SerializedName("dailyPnL")
    val dailyPnL: Double,
    
    @SerializedName("totalPnL")
    val totalPnL: Double
)